apps/explorer/src/ui/TransactionCard.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChevronRight12, ChevronRight16 } from '@mysten/icons';
+import clsx from 'clsx';
+import { type ReactNode, useState } from 'react';
+
+import { Card, type CardProps } from '~/ui/Card';
+import { Divider } from '~/ui/Divider';
+import { Heading } from '~/ui/Heading';
+import { Text } from '~/ui/Text';
+
+type Size = 'md' | 'sm';
+
+interface TransactionCardSectionProps {
+    children: ReactNode;
+    collapsedOnLoad?: boolean;
+    title?: string | ReactNode;
+}
+
+export function TransactionCardSection({
+    title,
+    collapsedOnLoad,
+    children,
+}: TransactionCardSectionProps) {
+    const [expanded, setExpanded] = useState<boolean>(!collapsedOnLoad);
+
+    const toggleExpanded = () => setExpanded((prevExpanded) => !prevExpanded);
+
+    return (
+        <div className="flex w-full flex-col gap-3">
+            {title && (
+                <div
+                    role="button"
```

Comment:
Use a `button` instead, or maybe even a `Disclosure` from headless UI (that's probably the right thing)